cdap-client/src/main/java/co/cask/cdap/client/AuthorizationClient.java

```diff
@@ -202,10 +202,12 @@ private HttpResponse executePrivilegeRequest(EntityId entityId, Principal princi
     FeatureDisabledException, UnauthenticatedException, IOException, NotFoundException, UnauthorizedException {
     HttpResponse httpResponse = doExecuteRequest(request, HttpURLConnection.HTTP_NOT_FOUND);
     if (HttpURLConnection.HTTP_NOT_FOUND == httpResponse.getResponseCode()) {
-      if (!listAllRoles().contains(new Role(principal.getName()))) {
-        throw new NotFoundException("Role: " + principal.getName());
+      if (principal.getType().equals(Principal.PrincipalType.ROLE) &&
+        httpResponse.getResponseMessage().equals(String.format("%s not found.", principal.getName()))) {
```

Comment:
But the error message "%s not found." comes from RoleNotFound exception. For other principal type, the error message might not match it. That's why I also check principal type to be Role first.  

Add a comment in the code? Or in this PR?